numba/targets/base.py

```diff
@@ -431,8 +431,12 @@ def get_constant_struct(self, builder, ty, val):
                       for i, v in enumerate(val)]
             return Constant.struct(consts)
 
+        elif isinstance(ty, types.Record):
+            consts = [self.get_constant(types.int8, b) for b in val.tobytes()]
```

Comment:
tobytes() gives you a str object under Python 2, and iterating gives you one-character strings.
Also, tobytes() appeared in Numpy 1.9, you should use tostring() instead (which does the same thing).
You can for example cast to bytearray:
```
>>> rec_X = np.recarray(10, dtype=np.dtype([('a', np.int32), ('b', np.float32)]))
>>> list(bytearray(rec_X.tostring()))
[224, 24, 169, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 234, 167, 1, 0, 0, 0, 0, 144, 243, 94, 1, 0, 0, 0, 0, 112, 58, 176, 1, 0, 0, 0, 0, 208, 70, 180, 1, 0, 0, 0, 0, 46, 105, 110, 116, 51, 50, 41, 44, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 111, 97, 116, 51, 128, 53, 134, 90, 244, 127, 0, 0]
```